packages/ckeditor5-ui/src/button/listitembuttonview.ts

```diff
@@ -0,0 +1,195 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module ui/button/listitembuttonview
+ */
+
+import type { ObservableChangeEvent, Locale } from '@ckeditor/ckeditor5-utils';
+import type ButtonLabel from './buttonlabel.js';
+import type ViewCollection from '../viewcollection.js';
+
+import { icons } from '@ckeditor/ckeditor5-core';
+import ButtonView from './buttonview.js';
+import ButtonLabelView from './buttonlabelview.js';
+import IconView from '../icon/iconview.js';
+import View from '../view.js';
+
+import '../../theme/components/button/listitembutton.css';
+
+/**
+ * Button that is used as dropdown list item entry.
+ */
+export default class ListItemButtonView extends ButtonView {
+	/**
+	 * Holds the view for the check icon of a button list item.
+	 */
+	private readonly _checkIconHolderView = new CheckIconHolderView();
+
+	/**
+	 * The flag that indicates if the button should render a check holder.
+	 *
+	 * @internal
+	 * @readonly
+	 * @observable
+	 */
+	declare public _shouldRenderCheckHolder: boolean;
+
+	/**
+	 * Indicates whether the button view has reserved space for a check holder.
+	 *
+	 * @observable
+	 */
+	declare public hasReservedCheckHolderSpace: boolean;
+
+	/**
+	 * @inheritDoc
+	 */
+	constructor( locale?: Locale, labelView: ButtonLabel = new ButtonLabelView() ) {
+		super( locale, labelView );
+
+		this.set( {
+			hasReservedCheckHolderSpace: false,
+			_shouldRenderCheckHolder: this.isToggleable
+		} );
+
+		const bind = this.bindTemplate;
+
+		this.extendTemplate( {
+			attributes: {
+				class: [
+					'ck-list-item-button',
+					bind.if( 'isToggleable', 'ck-list-item-button_toggleable' )
+				]
+			}
+		} );
+
+		this.bind( '_shouldRenderCheckHolder' ).to(
+			this, 'hasReservedCheckHolderSpace',
+			this, 'isToggleable',
+			( hasReservedCheckHolderSpace, isToggleable ) => hasReservedCheckHolderSpace || isToggleable
+		);
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public override render(): void {
+		super.render();
+
+		if ( this._shouldRenderCheckHolder ) {
+			this.children.add( this._checkIconHolderView, 0 );
+		}
+
+		this._watchCheckIconHolderMount();
+	}
+
+	/**
+	 * Renders the check icon if the button is toggleable.
+	 */
+	private _watchCheckIconHolderMount() {
+		this._checkIconHolderView
+			.bind( 'isOn' )
+			.to( this, 'isOn', value => this.isToggleable && value );
+
+		this.on<ObservableChangeEvent<boolean>>(
+			'change:_shouldRenderCheckHolder',
+			( evt, propertyName, shouldRenderCheckHolder ) => {
+				const { children, _checkIconHolderView } = this;
+
+				if ( shouldRenderCheckHolder && !children.has( _checkIconHolderView ) ) {
+					children.add( _checkIconHolderView, 0 );
+				} else if ( !shouldRenderCheckHolder && children.has( _checkIconHolderView ) ) {
+					children.remove( _checkIconHolderView );
+				}
```

Comment:
> Do you have 100% CC? Doesn't it complain about else path?

Yes, and it's tested here:
https://github.com/ckeditor/ckeditor5/blob/ad98d1fcfdec8b53f4ca3c3cedfd4356b1a0b6c1/packages/ckeditor5-ui/tests/button/listitembuttonview.js#L87-L94

> I ask because since you listen to change on boolean you have only two options false -> true or true -> false. So, based on new value, previous value is always deterministic and same for children state, right?

You are correct, removed.